### PR TITLE
Use axios and more safety around file uploads

### LINKED ISSUE #465: When bridging files to Matrix from Slack, some requests can fail authentication and upload a slack login page
See https://github.com/matrix-org/matrix-appservice-slack/blob/5cd8511269077f39ac83090cac2c07aada93de62/src/BridgedRoom.ts#L751

### PR DESCRIPTION
**Pull request by Half-Shot**
Hopefully fixes #465

### CODE REVIEW

File: src/BridgedRoom.ts [hunk 1]
```diff
@@ -676,12 +676,36 @@ export class BridgedRoom {
...
         }
 
+        let sendAsLink = false;
+        const authToken = this.SlackClient?.token;
+
+        if (!authToken) {
+            log.error("We have no client (or token) that can handle this file, sending as link");
+            sendAsLink = true;
+        } else if (maxUploadSize && file.size > maxUploadSize) {
+            log.warn(`File size too large (${file.size / 1024}KB > ${maxUploadSize / 1024} KB)`);
```
Comment (jaller94) [inline src/BridgedRoom.ts:686]: KiB when dividing by 1024.
```suggestion
            log.warn(`File size too large (${file.size / 1024}KiB > ${maxUploadSize / 1024} KiB)`);
```

File: src/BridgedRoom.ts [hunk 2]
```diff
@@ -676,12 +676,36 @@ export class BridgedRoom {
             return;
         }
 
+        let sendAsLink = false;
+        const authToken = this.SlackClient?.token;
```
Comment (jaller94) [inline src/BridgedRoom.ts:680]: Did you mean to use the parameter `client` or why did you add that above?

File: src/BridgedRoom.ts (hunk 1 above)
Comment (Half-Shot) [reply src/BridgedRoom.ts:686]: :+1: i get those mixed up

File: src/BridgedRoom.ts (hunk 2 above)
Comment (Half-Shot) [reply src/BridgedRoom.ts:680]: I think the param was used originally then removed. Removing it now.

File: changelog.d/463.bugfix [hunk 3]
```diff
@@ -0,0 +1 @@
+Don't automatically join new users to all public channels
```
Comment (jaller94) [inline changelog.d/463.bugfix:1]: #463 is not this PR but #463 added `changelog.d/431.bugfix`... okaaaaaayyyy.

Comment (Half-Shot) [reply changelog.d/463.bugfix:1]: i think someone (me) might have left out a file from another PR :s

**Comment by williamkray** [comment]
i've just deployed the docker container including this fix and now _all_ image uploads from slack are corrupt in matrix. 

when downloading and opening the image from matrix (e.g. https://synapse.jobmachine.org/_matrix/media/r0/download/jobmachine.org/pSdoKVhARBgOFICbhfvGrjLk and https://synapse.jobmachine.org/_matrix/media/r0/download/jobmachine.org/iDKjBHZSrMGwXPTAHDYTkRFR are a few examples from my testing) the file either just says "Error serving file", or shows that it is an image data file, but no image programs can open it properly (chromium, GIMP, feh, etc). GIMP says "Not a JPEG file: starts with 0xef 0xbf"

this testing includes examples where i both waited for the image processing dialogs to finish in slack before sending to the channel, and tests where i did not. i've used several images of varying sizes, but i haven't had a single one come through to matrix yet.

**Comment by Half-Shot** [comment]
>i've just deployed the docker container including this fix and now all image uploads from slack are corrupt in matrix.

Sad :c. I'll take another look.

**Pull request merged**